File: A1.ts
/**
 * @author Marco Cantusci
 * @verion 1.0.0
 * @date 2025-12-04
 * @fileoverview Prompts the sum of integers that are in range and out of range for a given range.
 */


// declare variables
let numberNumber: number;
let inRange: number = 0;
let outRange: number = 0;


// asking for range
const startString: string = prompt("Enter an integer for the start of the range: ") || ("Invalid");
const startNumber: number = parseInt(startString);

const endString: string = prompt("Enter an integer for the start of the range: ") || ("Invalid");
const endNumber: number = parseInt(endString);

// do while loop
do {

  const numberString: string = prompt("Enter an integer or zero(0) to end: ") || ("Invalid") // asks for integer until 0 is pressed.
  numberNumber = parseInt(numberString);

  if (numberNumber >= startNumber && numberNumber <= endNumber) { // adds to total sum
    inRange += numberNumber;
  } else {
    outRange += numberNumber;
  }

} while (numberNumber !== 0); // print out sums
  console.log(`The sum of the integers inside the range is ${inRange}`);
  console.log(`The sum of the integers outside the range is ${outRange}`);

  console.log("\nDone.");

File: A1.go
/**
 * @author Marco Cantusci
 * @verion 1.0.0
 * @date 2025-12-04
 * @fileoverview Prompts the sum of integers that are in range and out of range for a given range.
 */

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	// declare variables
	var startString string
	var startNumber float64
	var endString string
	var endNumber float64
	var numberString string
	var numberNumber float64 = 1 // makes sure it runs once
	var inRange int
	var outRange int

	reader := bufio.NewReader(os.Stdin)

	// asking for range
	fmt.Print("Enter an integer for the start of the range: ")
	startString, _ = reader.ReadString('\n')
	startString = strings.TrimSpace(startString)
	startNumber, _ = strconv.ParseFloat(startString, 64)

	fmt.Print("Enter an integer for the end of the range: ")
	endString, _ = reader.ReadString('\n')
	endString = strings.TrimSpace(endString)
	endNumber, _ = strconv.ParseFloat(endString, 64)

	// do while loop
	for int(numberNumber) != 0 {

		fmt.Print("Enter an integer or zero(0) to end: ") // Asks for integers until 0 is entered.
		numberString, _ = reader.ReadString('\n')
		numberString = strings.TrimSpace(numberString)
		numberNumber, _ = strconv.ParseFloat(numberString, 64)

		if int(numberNumber) >= int(startNumber) && int(numberNumber) <= int(endNumber) { // adds to the total sum
			inRange += int(numberNumber)
		} else {
			outRange += int(numberNumber)
		}
	}

	fmt.Printf("The sum of the integers inside the range is %d \n", inRange)
	fmt.Printf("The sum of the integers outside the range is %d \n", outRange)

	fmt.Println("\nDone.")
}
